packages/ui/src/components/PropertiesModal/PropertiesModal.tsx

```diff
@@ -1,33 +1,141 @@
 import { Modal } from '@patternfly/react-core';
-import { Caption, Table, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
 import { FunctionComponent } from 'react';
-import { camelComponentToTable, kameletToTable } from '../../camel-utils/camel-to-table.adapter';
+import {
+  camelComponentApisToTable,
+  camelComponentPropertiesToTable,
+  camelProcessorPropertiesToTable,
+  kameletToPropertiesTable,
+} from '../../camel-utils/camel-to-table.adapter';
 import { CatalogKind, ICamelComponentDefinition, ICamelProcessorDefinition, IKameletDefinition } from '../../models';
 import { ITile } from '../Catalog';
 import { EmptyTableState } from './EmptyTableState';
-import { IPropertiesTable } from './PropertiesModal.models';
+import { IPropertiesTab } from './PropertiesModal.models';
 import './PropertiesModal.scss';
+import { PropertiesTabs } from './PropertiesTabs';
+import { transformPropertiesIntoTab } from './camel-properties-to-tab';
 
 interface IPropertiesModalProps {
   tile: ITile;
   onClose: () => void;
   isModalOpen: boolean;
 }
 
+const transformCamelComponentIntoTab = (
```

Comment:
Could we please place these functions in a separate file?

Optional: Maybe as static methods of a class?